homeassistant/components/onkyo/media_player.py

```diff
@@ -506,16 +522,12 @@ def _parse_video_inforamtion(self, video_information_raw):
         else:
             self._attributes.pop(ATTR_VIDEO_INFORMATION, None)
 
+    def _query_delayed_av_info(self):
+        if not self._query_timer:
+            self._query_timer = self.hass.loop.call_later(10, self._query_av_info)
```

Comment:
This does create a potential bug - if you change stuff quickly enough, the second change won't trigger a query, and the first one might be too quick and pick up on the old values or temp ones. I think we can live with it, though.

Reply:
Well the problem I had with the timer being reset every time was that when the screen changed too often, the _query_av_info was never triggered. That's why I now only query a new one when the timer has finished. But if you know a better solution I'd like to hear it!

Reply:
Yeah, I think your approach makes sense, and is probably better than mine. Don't have a good way to resolve this. It's a shame the AVR doesn't trigger the updates correctly by itself.